core/src/epicli/data/common/ansible/playbooks/roles/applications/tasks/deploy-k8s-app.yml

```diff
@@ -1,24 +1,26 @@
 ---
-- name: "{{ data.name }}: Set directory for deployment files"
-  set_fact:
-    deployment_tmp_dir: /tmp/epiphany/apps/{{ app_dir_name }}
+- name: Assert input parameters
+  assert:
+    that:
+      - deployment_tmp_root_dir is defined
+      - app_dir_name is defined
 
-- name: "{{ data.name }}: Create directory for deployment files"
-  file:
-    path: "{{ deployment_tmp_dir }}"
-    state: directory
-    owner: "{{ admin_user.name }}"
-    group: "{{ admin_user.name }}"
-
-- name: "{{ data.name }}: Deploy K8s files"
-  include_tasks: deploy-k8s-file.yml
+- name: Deploy application (k8s)
   vars:
-    file_name: "{{ deployment_file }}"
-  loop: "{{ deployment_files }}"
-  loop_control:
-    loop_var: deployment_file
+    deployment_tmp_dir: "{{ deployment_tmp_root_dir }}/{{ app_dir_name }}"
+  always:
+    - name: "{{ data.name }}: Remove temporary directory"
+      file:
+        path: "{{ deployment_tmp_dir }}"
+        state: absent
+  block:
+    - name: "{{ data.name }}: Create directory for deployment files"
+      file:
+        path: "{{ deployment_tmp_dir }}"
+        state: directory
 
-- name: "{{ data.name }}: Remove temporary directory"
-  file:
-    path: "{{ deployment_tmp_dir }}"
-    state: absent
+    - name: "{{ data.name }}: Deploy K8s files"
+      include_tasks: deploy-k8s-file.yml
```

Comment:
May be it makes sense to place all files into a directory and then execute `kubectl apply -f <dir>` only once for this directory. WDYT?

Reply:
This is actually a good point. Let me work on it :+1:😇

Reply:
Please take a look at latest commits. :+1: